### DISCUSSION TITLE
WebView2 PDF spacing.

### DISCUSSION THREAD

**Discussion by Petrarca181** [microsoft/microsoft-ui-xaml#9748] [Q&A]
Hello, there is a way to make WV2  to set desired spacing between pdf pages? I want to make it look like in summary window.  
Also how to enable full screen mode from code  behind?

**Comment by castorix** [comment]
I don't see which result you want to get but otherwise you can render PDF with [PdfCreateRenderer ](https://learn.microsoft.com/en-us/windows/win32/api/windows.data.pdf.interop/nf-windows-data-pdf-interop-pdfcreaterenderer) and then customize the resulting images

**Reply by Petrarca181** [reply]
> I don't see which result you want to get but otherwise you can render PDF with [PdfCreateRenderer ](https://learn.microsoft.com/en-us/windows/win32/api/windows.data.pdf.interop/nf-windows-data-pdf-interop-pdfcreaterenderer) and then customize the resulting images

I need fast .pdf and .ai files preview.  Render pdf to images is not the best solution for me, sometimes there can be around 100 artboards. I just need  more space here between 2 pages for better document readability. 
![image](https://github.com/microsoft/microsoft-ui-xaml/assets/54235138/b96facc7-a8ad-4c66-a30c-e2e53b09656a)

**Comment by castorix** [comment]
> I need fast .pdf and .ai files preview

_IPdfRendererNative::RenderPageToDeviceCont_ is fast as it uses GPU
But I don't know for .ai files  (I did a quick test with a WebView2/VB.NET and it does not seem to read .ai files (Windows 10 22H2)...)

**Reply by Petrarca181** [reply]
> > I need fast .pdf and .ai files preview
> 
> _IPdfRendererNative::RenderPageToDeviceCont_ is fast as it uses GPU But I don't know for .ai files (I did a quick test with a WebView2/VB.NET and it does not seem to read .ai files (Windows 10 22H2)...)

Office pc with limited ram and integrated cpu. As I said rendering is not a solution. .ai illustrator file is a postscript with some metatada. Just rename it to pdf and any pdf reader will open it. 

I hoped for some quick solution like ExecuteScriptAsync("document.pages.margin = 10").  Still not a big problem.  With WebView2 and 10 lines of code I achived 99% of what I need.

**Reply by castorix** [reply]
> Office pc with limited ram and integrated cpu. As I said rendering is not a solution. .ai illustrator file is a postscript with some metatada. Just rename it to pdf and any pdf reader will open it.

(I have also a PC without real graphic card, but all interfaces related to Direct2D work fine)
Yes, I tested with some .ai file from Google, but there are different formats and  I cannot open some
For example, this small one : https://app.box.com/s/ixknwbbozdxgueumxenoyt02hxy0manx , I can only open it with Adobe apps)

**Reply by NicholasChrzan** [reply]
Since you are using WV2 you could render the PDF via PDF.js and there are quite a bit of formatting options available in that library.
